pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/dao/util/DaoUtil.java

```diff
@@ -0,0 +1,60 @@
+package org.broadinstitute.dsm.db.dao.util;
+
+import static org.broadinstitute.ddp.db.TransactionWrapper.inTransaction;
+
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.Optional;
+
+import lombok.extern.slf4j.Slf4j;
+import org.broadinstitute.lddp.db.SimpleResult;
+
+@Slf4j
+public class DaoUtil {
+
+    private DaoUtil() {}
+
+    /**
+     * Get a record by ID
+     *
+     * @param id record ID, must be in first substitution position of query
+     * @param query get query
+     * @return SimpleResult with Optional result or exception
+     */
+    public static SimpleResult getById(long id, String query, ResultsBuilder resultsBuilder) {
+        return inTransaction(conn -> {
+            SimpleResult dbVals = new SimpleResult(Optional.empty());
+            try (PreparedStatement stmt = conn.prepareStatement(query)) {
+                stmt.setLong(1, id);
+                try (ResultSet rs = stmt.executeQuery()) {
+                    if (rs.next()) {
+                        dbVals.resultValue = Optional.of(resultsBuilder.build(rs));
+                    }
+                }
+            } catch (SQLException e) {
+                dbVals.resultException = e;
+            }
+            return dbVals;
+        });
+    }
+
+    /**
+     * Delete a record by ID
+     *
+     * @param id record ID, must be in first substitution position of query
+     * @param query delete query
+     */
+    public static SimpleResult deleteById(int id, String query) {
+        return inTransaction(conn -> {
+            SimpleResult res = new SimpleResult();
+            try (PreparedStatement stmt = conn.prepareStatement(query)) {
```

Comment:
I like what you've got going on in this utility!

I can't remember if ParameterMetaData is available in our version of MySQL.  If it is, it could be used to enforce the "must be in first substitution position of query" rule you've got noted in the comments.  It may also help to test to make sure there's only one parameter required in the query.  No impact required for this PR.

Reply:
I am using metadata elsewhere, so I'll see if I can apply in here in a future PR.
Thx for the suggestion!

Reply:
I am using metadata elsewhere in the PR, so I see if there is a way apply it here.
Thx for the suggestion!